### DISCUSSION TITLE
RC_REF_COMPARISON - default suspicious types

### DISCUSSION THREAD

**Discussion by yuliyasokh** [spotbugs/spotbugs#2532] [General]
Hello All, 
While investigating https://github.com/spotbugs/spotbugs/issues/2297 I found there's pre-defined list of classes DEFAULT_SUSPICIOUS_SET that are suspicious if compared by reference: 
 `

    static {
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Boolean");
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Byte");
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Character");
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Double");
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Float");
        DEFAULT_SUSPICIOUS_SET.add(Values.DOTTED_JAVA_LANG_INTEGER);
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Long");
        DEFAULT_SUSPICIOUS_SET.add("java.lang.Short");
    }`

What do you think about removing this list and fire a bug if **any** non primitive is compared using == operator?

**Comment by ThrawnCA** [comment]
Enums would need to be an exception. Comparing them with `==` is valid and often recommended.

**Comment by iloveeclipse** [comment]
> What do you think about removing this list and fire a bug if **any** non primitive is compared using == operator?

Try to run this proposal on few projects of reasonable size and check false positives rate compared to real issues found.
While references check is seldom, there could be valid reasons to have it.
